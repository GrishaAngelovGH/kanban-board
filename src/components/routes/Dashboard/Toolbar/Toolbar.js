const Toolbar = ({ items, onClick }) => (
  <div className="mt-2 row g-0 justify-content-center">
    <div className="col-11 bg-primary-subtle rounded p-2">
      <p className="text-secondary text-center">Click on any item to add the corresponding card to the dashboard</p>
      <div className="row flex-column flex-md-row g-1 justify-content-evenly">
        {
          items.map((v, i) => (
            <div
              key={i}
              data-testid="toolbar-item"
              className="col-md-2 bg-primary text-white text-center rounded"
              onClick={() => { onClick(v.id) }}
            >
              {v.title}
            </div>
          ))
        }
      </div>
    </div>
  </div>
)

export default Toolbar