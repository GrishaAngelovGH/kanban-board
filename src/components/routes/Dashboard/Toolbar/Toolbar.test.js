import { render } from "@testing-library/react"

import Toolbar from "./Toolbar"

test("should render Dashboard/Toolbar component", () => {
  const view = render(
    <Toolbar items={[{ title: "Item 1" }, { title: "Item 2" }]} />
  )

  expect(view).toMatchSnapshot()
})