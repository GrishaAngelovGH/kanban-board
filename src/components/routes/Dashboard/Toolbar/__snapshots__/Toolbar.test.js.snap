// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dashboard/Toolbar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-2 row g-0 justify-content-center"
      >
        <div
          class="col-11 bg-primary-subtle rounded p-2"
        >
          <p
            class="text-secondary text-center"
          >
            Click on any item to add the corresponding card to the dashboard
          </p>
          <div
            class="row flex-column flex-md-row g-1 justify-content-evenly"
          >
            <div
              class="col-md-2 bg-primary text-white text-center rounded"
            >
              Item 1
            </div>
            <div
              class="col-md-2 bg-primary text-white text-center rounded"
            >
              Item 2
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-2 row g-0 justify-content-center"
    >
      <div
        class="col-11 bg-primary-subtle rounded p-2"
      >
        <p
          class="text-secondary text-center"
        >
          Click on any item to add the corresponding card to the dashboard
        </p>
        <div
          class="row flex-column flex-md-row g-1 justify-content-evenly"
        >
          <div
            class="col-md-2 bg-primary text-white text-center rounded"
          >
            Item 1
          </div>
          <div
            class="col-md-2 bg-primary text-white text-center rounded"
          >
            Item 2
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
